hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -770,6 +785,12 @@ static QString* psh_convert(struct PixelShader *ps)
     QString *final = qstring_new();
     qstring_append(final, "#version 330\n\n");
     qstring_append(final, qstring_get_str(preflight));
+    qstring_append(final, "\n");
+    qstring_append(final, "vec4 signed_texture(vec4 sample, bvec4 mask) {\n");
+    qstring_append(final, "  vec4 signed_sample = sample * 2.0 - 1.0;\n");
+    qstring_append(final, "  return mix(sample, signed_sample, mask);\n");
```

Comment:
```
// Hack so interpolation will work properly
uint8_biased = uint8 xor 0x80;

// Normalization step
//FIXME: GL will use / 255.0, or / 15.0 etc.?
uint8_biased_normalized = uint8_biased/254;

// Unbias to recover intended value
sint8 = (uint8_biased_normalized*2-1)-2/254
```

is closer to values shown for SNORM8 at https://github.com/nlguillemot/SNormTest

Reply:
Also see equation 2.3 (page 11) of https://www.khronos.org/registry/OpenGL/specs/gl/glspec33.core.pdf